src/vsr/replica.zig

```diff
@@ -5266,8 +5270,46 @@ pub fn ReplicaType(
             });
         }
 
+        fn send_reply_message_to_client(self: *Self, reply: *Message) void {
+            assert(reply.header.command == .reply);
+            assert(reply.header.view <= self.view);
+
+            // If request committed in a different view than the one it was originally submitted
+            // with, we must inform the client about this newer view before we send it a reply.
```

Comment:
Nit: "originally submitted with" should be "originally prepared in". "Originally submitted" implies that what matters is when the client first sent the request, but what actually matters is when it arrived at the replica.

Also, "If _the_ request".